### PR TITLE
Docs: Cleanup & update to developer-related docs

### PR DESCRIPTION
**Pull request by brbrr**
#### Changes proposed in this Pull Request:

* documentation updates

#### Testing instructions:

* read the docs
* check for typos

#### Proposed changelog entry for your changes:

- ⛔️

### CODE REVIEW

File: docs/development-environment.md [hunk 1]
```diff
@@ -2,10 +2,15 @@
 
 The javascript and CSS components of this plugin's admin interface need to be built in order to get the runtime bundle (`_inc/build/admin.js`)
 
-**Recommended Environment**
+## Before moving forward
+
+In most cases you want to have accessible WordPress installation for Jetpack development. We suggest to use `Docker` container which we provide. Follow [this guide](../docker/README.md#to-get-started) to configure your Docker development environment.
```
Comment (jeherve) [inline docs/development-environment.md]: "you want to have accessible **an** WordPress installation", maybe?

Comment (jeherve) [inline docs/development-environment.md]: "We suggest to use **a** `Docker` container", maybe?

File: docs/development-environment.md [hunk 2]
```diff
@@ -2,10 +2,15 @@
...
+
+**Recommended Environment:**
 
 * Node.js 10
 * Yarn 1.7
+* PHP 7.0 (in case you running WordPress locally)
```
Comment (jeherve) [inline docs/development-environment.md]: "you **are** running WordPress"

**Review by jeherve — COMMENTED** [review]
👍 

Any specific reasons why you removed those 2 documents?

**Review by kraftbj — APPROVED** [review]

**Pull request merged**